src/js/views/generate.js

```diff
@@ -103,6 +115,55 @@
             });
         },
 
+        /**
+         * Generates records automatically populated with dummy data
+         * (using faker.js)
+         */
+        generateRecordsAuto: function() {
+            var module = this.$('#generateAuto, #generateManual').not('.hide').find('select[name=Module]').val();
+            var numberOfRecords = this.$('#generateAuto, #generateManual').not('.hide').find('input[name=numberOfRecords]').val();
+
+            BDT.page.eval('getModuleFields', [module], (fields, isException) => {
+                if (isException) {
+                    BDT.page.eval('console', ['error', '`getModuleFields` could not get the results.']);
+                } else {
+                    if (fields) {
+                        let attributesArray = [];
+                        for (let i=0; i < numberOfRecords; i++) {
+                            let attributes = {};
+                            _.each(fields, (meta, fieldName) => {
+                                let value;
+
+                                // Set the specific value if we have one defined
+                                // in the data-mapper.
+                                if (BDT.dataMap.fixedValues[module] && BDT.dataMap.fixedValues[module][fieldName]) {
+                                    value = BDT.dataMap.fixedValues[module][fieldName];
+                                    // Else, generate a value using faker.js.
+                                } else {
+                                    let fakerField = BDT.dataMap.fieldNames[fieldName];
+                                    // If the field name is not defined in the
+                                    // mapping, fallback to the 'type' mapping.
+                                    if (_.isEmpty(fakerField) && !BDT.dataMap.blackList[fieldName]) {
```

Comment:
can't we have a list of blacked list fields and then just omit them from fields on line 130?
even in terms of performance it avoids looping on fields that will be skipped